core/src/main/java/org/apache/calcite/util/mapping/Mappings.java

```diff
@@ -434,7 +434,9 @@ public static boolean keepsOrdering(TargetMapping mapping) {
       if (target != -1 && target < prevTarget) {
         return false;
       }
-      prevTarget = target;
+      if (target != -1) {
```

Comment:
Maybe we can combine two `if` clause together?

Reply:
Sure